eva-accession-clustering-automation/clustering_automation/create_clustering_progress_table.py

```diff
@@ -70,218 +69,213 @@ def create_table(private_config_xml_file):
-        for taxonomy, scientific_name, assembly_accession, source, num_ss_id \
-                in get_all_results_for_query(pg_conn, query_origin_asm):
-            insert_remapping_entries(pg_conn, taxonomy, scientific_name, assembly_accession, source,
-                                     num_ss_id, False, reference_directory, release_version)
-
-
-def insert_remapping_entries(pg_conn, taxonomy, scientific_name, assembly_accession, source, num_ss_id,
-                             should_be_cls, reference_directory, release_version):
-    if num_ss_id == 0:  # Do not release species with no data
-        return
-    should_be_clustered = should_be_cls
-    should_be_released = True
-    ncbi_assembly = NCBIAssembly(assembly_accession, scientific_name, reference_directory)
-    fasta_path = ncbi_assembly.assembly_fasta_path
-    report_path = ncbi_assembly.assembly_report_path
-    tempmongo_instance = get_tempmongo_instance(pg_conn, taxonomy)
-    release_folder_name = normalise_taxon_scientific_name(scientific_name)
-    query_insert = (
-        'INSERT INTO eva_progress_tracker.clustering_release_tracker '
-        '(sources, taxonomy, scientific_name, assembly_accession, release_version, should_be_clustered, '
-        'fasta_path, report_path, tempmongo_instance, should_be_released, release_folder_name) '
-        f"VALUES ('{source}', {taxonomy}, '{scientific_name}', '{assembly_accession}', {release_version}, "
-        f"{should_be_clustered}, '{fasta_path}', '{report_path}', '{tempmongo_instance}', {should_be_released}, "
-        f"'{release_folder_name}') ON CONFLICT DO NOTHING")
-    execute_query(pg_conn, query_insert)
-
-
-def fill_in_from_previous_release(private_config_xml_file, curr_release_version):
-    query = f"""select taxonomy,scientific_name,assembly_accession,sources,fasta_path,report_path,
-    release_folder_name from eva_progress_tracker.clustering_release_tracker 
-            where release_version = {curr_release_version - 1}"""
+def get_scientific_name(pg_conn, taxonomy):
+    query = f'SELECT scientific_name from evapro.taxonomy where taxonomy_id={taxonomy}'
+    results = get_all_results_for_query(pg_conn, query)
+    if not results:
+        raise Exception(f'No entry found in taxonomy table for taxonomy: {taxonomy}')
+    sc_name = results[0][0]
+    if "'" in sc_name:
+        # some scientific names in db has single quote (') in name causing issues while inserting
+        sc_name = sc_name.replace("'", "\''")
+        return sc_name
+    else:
+        return sc_name
+
+
+def fill_in_from_previous_release(private_config_xml_file, profile, curr_release_version):
+    query = f"""select taxonomy, scientific_name, assembly_accession, sources, fasta_path, report_path, 
+                release_folder_name from eva_progress_tracker.clustering_release_tracker 
+                where release_version = {curr_release_version - 1} and assembly_accession != 'Unmapped'"""
```

Comment:
Why are we not propagating the unmapped ?

Reply:
updated